lib/components/user/nav-login-button.js

```diff
@@ -78,21 +79,25 @@ export default class NavLoginButton extends Component {
 
           {links && links.map((link, i) => (
             <LinkContainerWithQuery exact key={i} target={link.target} to={link.url}>
-              <MenuItem>{link.text}</MenuItem>
+              <MenuItem>
+                <FormattedMessage id={link.messageId} />
```

Comment:
Please do not use variables as message IDs as it is hard to automatically keep track of which IDs are in the code. Instead, please create a hardcoded message ID string for each possible message.